File: spring-reactive-02/learn-java/src/main/java/boki/learnjava/v3_transformsequence/a_map/MapEx1.java
package boki.learnjava.v3_transformsequence.a_map;

import boki.learnjava.utils.Logger;
import reactor.core.publisher.Flux;

/**
 * map 기본 개념 예제
 *  - Upstream에서 emit된 데이터를 반환한 후, DownStream으로 emit
 */
public class MapEx1 {
    public static void main(String[] args) {
        Flux
            .just("Green-Circle", "Yellow-Circle", "Blue-Circle", "Red-Circle")
            .map(circle -> circle.replace("Circle", "Rectangle"))
            .subscribe(Logger::onNext);
    }
}

File: spring-reactive-02/learn-kt/src/main/kotlin/boki/learnkt/v3_transformsequence/a_map/MapEx1.kt
package boki.learnkt.v3_transformsequence.a_map

import boki.learnkt.util.Logger
import reactor.core.publisher.Flux

/**
 * map 기본 개념 예제
 *  - Upstream에서 emit된 데이터를 반환한 후, DownStream으로 emit
 */
fun main() {
   Flux
       .just("Green-Circle", "Yellow-Circle", "Blue-Circle", "Red-Circle")
       .map { it.replace("Circle", "Rectangle") }
       .subscribe { Logger.onNext(it) }
}